product-scenarios/6-rollout-api-updates-seamlessly/6.1-rollout-api-updates-to-existing-api/6.1.1-update-published-api-using-api-publisher/src/test/java/org/wso2/am/scenario/tests/update/published/api/using/publisher/UpdatePublishedAPITest.java

```diff
@@ -0,0 +1,228 @@
+        String provider = CREATOR_USERNAME;
+        String apiName = UUID.randomUUID().toString();
+        String apiVersion = "1.0.0";
+        String endpointUrl = "http://test";
+
+        Swagger2Builder swagger = buildSwagger(apiName, apiVersion);
+
+        apiPublisher.login(CREATOR_USERNAME, CREATOR_PASSWORD);
+
+        APIIdentifier apiIdentifier = addAPI(provider, apiName, apiVersion, endpointUrl, swagger);
+
+        apiPublisher.login(PUBLISHER_USERNAME, PUBLISHER_PASSWORD);
+
+        publishAPI(apiIdentifier);
+
+        // Get published API in store
+        HttpResponse apiResponse = apiStore.getAPI(provider, apiName, apiVersion);
+
+        JSONObject api = new JSONObject(apiResponse.getData());
+
+        JSONArray apiTemplates = api.getJSONObject(API_KEY).getJSONArray(TEMPLATES_KEY);
+        String apiResources = api.getJSONObject(API_KEY).getString(RESOURCES_KEY);
+
+        Assert.assertEquals(apiTemplates.length(), 2);
+        Assert.assertTrue(apiResources.contains(ADD_RESOURCE));
+        Assert.assertTrue(apiResources.contains(RETRIEVE_RESOURCE));
+
+        // Add a new resource to existing swagger
+        addNewResourceToSwagger(swagger);
+
+        // Update API
+        updateAPI(apiName, apiVersion, endpointUrl, swagger);
+
+        // Check new resource in store
+        HttpResponse updatedApiResponse = apiStore.getAPI(provider, apiName, apiVersion);
+
+        JSONObject updatedApi = new JSONObject(updatedApiResponse.getData());
+
+        JSONArray updatedApiTemplates = updatedApi.getJSONObject(API_KEY).getJSONArray(TEMPLATES_KEY);
+        String updatedApiResources = updatedApi.getJSONObject(API_KEY).getString(RESOURCES_KEY);
+
+        Assert.assertEquals(updatedApiTemplates.length(), 3);
+        Assert.assertTrue(updatedApiResources.contains(ADD_RESOURCE));
+        Assert.assertTrue(updatedApiResources.contains(RETRIEVE_RESOURCE));
+        Assert.assertTrue(updatedApiResources.contains(UPDATE_RESOURCE));
+
+        int matchingTemplateCount = 0;
+        JSONArray newTemplate = new JSONArray();
+
+        for (int i = 0; i < apiTemplates.length(); ++i) {
+            //JSONString apiTemplate = (JSONString) ((JSONArray)apiTemplates.get(0)).get(i);
+            JSONArray apiTemplate = (JSONArray) apiTemplates.get(i);
+
+            for (int j = 0; j < updatedApiTemplates.length(); ++j) {
+                JSONArray updatedApiTemplate = (JSONArray) updatedApiTemplates.get(j);
+
+                if (apiTemplate.toString().equals(updatedApiTemplate.toString())) {
+                    ++matchingTemplateCount;
+                    break;
+                } else {
+                    newTemplate = updatedApiTemplate;
+                }
+            }
+        }
+
+        Assert.assertEquals(matchingTemplateCount, 2);
+        Assert.assertTrue(newTemplate.toString().contains(UPDATE_RESOURCE));
+
+
+        System.out.println(updatedApi);
```

Comment:
We have been using a log4j.properties for each module, and logging using that in other modules. Shall we do the same?